game/src/ecs/components/ai/transition/ProtectOnAttack.java

```diff
@@ -0,0 +1,66 @@
+package ecs.components.ai.transition;
+
+import ecs.components.HealthComponent;
+import ecs.components.MissingComponentException;
+import ecs.components.PlayableComponent;
+import ecs.entities.Entity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Lets AI protect an entity
+ *
+ * <p>Implements an AI that protects a specific entity with a HealthComponent if the hero afflicting damage to it.
+ * Entity will stay in fight mode</p>
+ */
+public class ProtectOnAttack implements ITransition {
+
+    private boolean isInfight = false;
+
+    private final List<Entity> toProtect = new ArrayList<>();
+
+    /**
+     * Constructor for one entity to protect
+     *
+     * @param entity
```

Comment:
```suggestion
     * @param entity to protect
```

aber gerne noch etwas ausführlicher